sdcm/cluster_gce.py

```diff
@@ -42,25 +45,28 @@ def __init__(self, gce_instance, gce_service, credentials, parent_cluster,  # py
                                       base_logdir=base_logdir,
                                       node_prefix=node_prefix,
                                       dc_idx=dc_idx)
-
-        if cluster.TEST_DURATION >= 24 * 60:
-            self.log.info('Test duration set to %s. '
-                          'Tagging node with "keep-alive"',
-                          cluster.TEST_DURATION)
-            self._instance_wait_safe(self._gce_service.ex_set_node_tags,
-                                     self._instance, ['keep-alive'])
-        if "db" in self.name and cluster.Setup.KEEP_ALIVE_DB_NODES:
-            self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_DB_NODES)
-            self._instance_wait_safe(self._gce_service.ex_set_node_tags,
-                                     self._instance, ['keep-alive'])
-        if "loader" in self.name and cluster.Setup.KEEP_ALIVE_LOADER_NODES:
-            self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_LOADER_NODES)
+        if not cluster.Setup.REUSE_CLUSTER:
+            metadata = []
+            ex_metadata = {'Name': name,
+                           'NodeIndex': node_index,
+                           'NodeType': node_type}
+            ex_metadata = gce_create_metadata(ex_metadata)
+            if cluster.TEST_DURATION >= 24 * 60:
+                self.log.info('Test duration set to %s. '
+                              'Tagging node with "keep-alive"',
+                              cluster.TEST_DURATION)
+                metadata.append('keep-alive')
+            elif "db" in self.name and cluster.Setup.KEEP_ALIVE_DB_NODES:
+                self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_DB_NODES)
+                metadata.append('keep-alive')
+            elif "loader" in self.name and cluster.Setup.KEEP_ALIVE_LOADER_NODES:
+                self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_LOADER_NODES)
+                metadata.append('keep-alive')
+            elif "monitor" in self.name and cluster.Setup.KEEP_ALIVE_MONITOR_NODES:
+                self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_MONITOR_NODES)
+                metadata.append('keep-alive')
             self._instance_wait_safe(self._gce_service.ex_set_node_tags,
-                                     self._instance, ['keep-alive'])
-        if "monitor" in self.name and cluster.Setup.KEEP_ALIVE_MONITOR_NODES:
-            self.log.info('Keep cluster on failure %s', cluster.Setup.KEEP_ALIVE_MONITOR_NODES)
-            self._instance_wait_safe(self._gce_service.ex_set_node_tags,
-                                     self._instance, ['keep-alive'])
+                                     self._instance, metadata, ex_metadata)
```

Comment:
Hi, @dkropachev  I saw a new err, it seems caused by this change.

```
19:05:28  < t:2019-11-27 11:05:27,866 f:cluster_gce.py  l:104  c:sdcm.cluster         p:ERROR > Node rolling-upgrade-upgrade--centos-db-node-c3a24b48-0-1 [35.231.52.66 | 10.142.0.123] (seed: False): Call to method <bound method GCENodeDriver.ex_set_node_tags of <libcloud.compute.drivers.gce.GCENodeDriver object at 0x7fa218048438>> (retries: 3) failed: ex_set_node_tags() takes 3 positional arguments but 4 were given
```

Reply:
Reference:
https://libcloud.readthedocs.io/en/latest/compute/drivers/gce.html#libcloud.compute.drivers.gce.GCENodeDriver.ex_set_node_tags
https://libcloud.readthedocs.io/en/latest/_modules/libcloud/compute/drivers/gce.html#GCENodeDriver.ex_set_node_tags

```
ex_set_node_tags(node, tags)

Parameters: 
- node (Node) – Node objecttags (list of str) 
– List of tags to apply to the object
```

Reply:
Seems like we missed that metadata has its own API call

It should be separated into tags, and metadata
https://libcloud.readthedocs.io/en/latest/compute/drivers/gce.html#libcloud.compute.drivers.gce.GCENodeDriver.ex_set_node_metadata

Reply:
Dmitry is working on fix